### DISCUSSION TITLE
fleet, the hive and other tools reset password

### DISCUSSION THREAD

**Discussion by pirke05** [Security-Onion-Solutions/securityonion#4455] [Unsupported Versions]
Hello,

How I can change the password for the tools I mentioned in subject name from the management console ? I used sudo so-user update name@domain.com, but it changed only for SOC and not for The Hive, Fleet etc. Please, i need fast help if possible :/

Best,

**Comment by pirke05** [comment]
any advice ?

**Reply by abesinger** [reply]
This isn't a complete answer, but see [#4452](https://github.com/Security-Onion-Solutions/securityonion/discussions/4452#discussioncomment-846914)

For FleetDM, If you can write bash & python & SQL, you can modify a copy of so-fleet-user-add to change the password; you'll see an (untested) example in my post. For TheHive, I don't know exactly, if you look at the code for so-thehive-user-add you can see it's a RESTful interface, maybe someone else can say what the REST is (pun intended)  for password changing. Or check TheHive documentation.

**Comment by Ron89** [comment]
To complete @asinger-ncat 's answer, let me help with TheHive:

For TheHive, you can create your script to change a user's password base on `so-thehive-user-add` script. But change the `curl` call line to the following:
```
 resp=$(curl -sk -XPOST -H "Authorization: Bearer $THEHIVE_KEY" -H "Content-Type: application/json" -L "https://$THEHVIE_API_URL/user/${THEHIVE_USER}/password/set" -d "{\"password\" : \"$THEHIVE_PASS\"}")
```

Note that this API call, if successful, will not give any response string, hence change the success condition to following:
```
 if [[ -z "$resp" ]]; then
     echo "Successfully changed TheHive user password"
 else
     echo "Changing TheHive user password failed"
     echo $resp
     exit 2
 fi
```
Hope it helps.

ref: http://docs.thehive-project.org/thehive/legacy/thehive3/api/user/
